src/java/org/apache/cassandra/cql3/terms/Tuples.java

```diff
@@ -63,6 +63,8 @@ public Literal(List<Term.Raw> elements)
 
         public Term prepare(String keyspace, ColumnSpecification receiver) throws InvalidRequestException
         {
+            validateTupleAssignableTo(receiver, elements);
```

Comment:
We just need to move this up a little bit. As of now, It is too late by the time we validate it.